java/src/main/native/src/TableJni.cpp

```diff
@@ -935,6 +935,39 @@ jlongArray combine_join_results(JNIEnv *env, cudf::table &left_results,
   return combine_join_results(env, std::move(left_cols), std::move(right_cols));
 }
 
+cudf::column_view remove_validity_from_col(cudf::column_view column_view) {
+  if (column_view.nullable() && column_view.null_count() == 0) {
+    // null_mask is allocated but no nulls present therefore we create a new column_view without the
+    // null_mask to avoid things blowing up in reading the parquet file
+    if (!cudf::is_nested(column_view.type())) {
```

Comment:
A child column may still need to be modified, even if a parent column is not nullable.